app/wasm/message_plugin.go

```diff
@@ -107,9 +108,17 @@ func performSwap(keeper *gammkeeper.Keeper, ctx sdk.Context, contractAddr sdk.Ac
 			PoolId:       swap.First.PoolId,
 			TokenInDenom: swap.First.DenomIn,
 		}}
+		output := swap.First.DenomOut
+		for _, step := range swap.Route {
+			routes = append(routes, gammtypes.SwapAmountOutRoute{
+				PoolId:       step.PoolId,
+				TokenInDenom: output,
+			})
+			output = step.DenomOut
```

Comment:
Ah, now I fully understand your idea.

Makes sense. My only concern would be, if this is fully clear / can be fully derived, or assumed, from the `ExactOut` specs.

Reply:
This is equivalent to the Osmosis API in that there is a 1:1 mapping of all valid inputs.
This one just seems more readable, consistent to me, in that you can switch between fixing input and fixing output very easily.